rust/src/passwd.rs

```diff
@@ -98,25 +98,27 @@ pub fn migrate_passwd_except_root(rootfs_dfd: i32) -> Result<()> {
         nameservice::passwd::parse_passwd_content(src_rd)?
     };
 
-    {
-        let mut etcdest_stream = rootfs.write_file(ETCTMP_PATH, 0o664).map(BufWriter::new)?;
-        let mut usrdest_stream = rootfs
-            .append_file(USRDEST_PATH, 0o664)
-            .map(BufWriter::new)?;
-
-        for user in entries {
-            let mut target = match user.uid {
-                0 => &mut etcdest_stream,
-                _ => &mut usrdest_stream,
-            };
-            user.to_writer(&mut target)?;
-        }
-
-        etcdest_stream.flush()?;
-        usrdest_stream.flush()?;
-    }
+    rootfs.write_file_with_sync(
+        ETCSRC_PATH,
+        0o664,
+        |mut etcdest_stream| -> std::io::Result<()> {
+            let mut usrdest_stream = rootfs
```

Comment:
This is again fine as is, but I do find it a bit odd how we're writing to another file inside the write handler for the first one.
Maybe cleaner to do something more like:
```
let (root, others): (Vec<_>, Vec<_>) = entries.iter().partition(|e| e.uid == 0);
// assert(root.len() == 1)
// write root entry
// write other entries?
```
?

Reply:
Indeed, good suggestion, does the version in the fixup commit look better?

Reply:
Yep looks great to me!

This is a great example of the "multi-paradigm" Rust where one can use more "functional" flows for parts like this ("partition" is a very functional thing) but use imperative where it makes sense (and it does in a lot).